homeassistant/components/reolink/host.py

```diff
@@ -0,0 +1,168 @@
+"""This component encapsulates the NVR/camera API and subscription."""
+from __future__ import annotations
+
+import asyncio
+import logging
+
+import aiohttp
+from reolink_ip.api import Host
+from reolink_ip.exceptions import (
+    ApiError,
+    CredentialsInvalidError,
+    InvalidContentTypeError,
+)
+
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import CONF_PROTOCOL, CONF_USE_HTTPS, DEFAULT_PROTOCOL, DEFAULT_TIMEOUT
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ReolinkHost:
+    """The implementation of the Reolink Host class."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: dict,
+        options: dict,
```

Comment:
We can probably narrow down typing to `dict[str, Any]` for `config` and `options`.